core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/entity/FromSolutionEntitySelector.java

```diff
@@ -147,6 +147,13 @@ private void checkCachedEntityListIsDirty() {
         }
     }
 
+    @Override
+    protected Object[] getEqualityRequirements() {
```

Comment:
Really feels like reinventing equals/hashcode.

And maybe error prone: if `AlphaFoo.getEqualityRequirements()` returns `{"one", 2}`
and `BetaFoo.getEqualityRequirements()` returns `{"one", 2}`
are they equal? Definitely not!

Reply:
In my opinion: everything that implements getEqualityRequirements() should just implement equals and hashcode.

To enforce them implementing it, you could add those methods as abstract methods on the interfaces/abstract classes. Doubt if I would, though. OTH, javadocs are weak indeed.